Disable NPT on RevA Barcelonas
This should only effect test boxes/SDVs

diff -r ee2e0f1176a9 xen/arch/x86/hvm/svm/svm.c
--- a/xen/arch/x86/hvm/svm/svm.c
+++ b/xen/arch/x86/hvm/svm/svm.c
@@ -1270,6 +1270,9 @@ const struct hvm_function_table * __init
         printk(" - none\n");
 
     svm_function_table.hap_supported = cpu_has_svm_npt;
+    /* NPT wasn't really usable on RevA Barcelonas (family 16 model 0) */
+    if ( (cpuid_eax(0x1) & 0xfffffff0) == 0x00100f00 )
+        svm_function_table.hap_supported = 0;
     svm_function_table.hap_capabilities = HVM_HAP_SUPERPAGE_2MB |
         ((cpuid_edx(0x80000001) & 0x04000000) ? HVM_HAP_SUPERPAGE_1GB : 0);
 
